zerver/lib/test_classes.py

```diff
@@ -113,6 +120,17 @@ def tearDown(self) -> None:
                 self.mock_ldap.reset()
             self.mock_initialize.stop()
 
+    def run(self, result: Optional[TestResult]=None) -> Optional[TestResult]:
+        if not settings.BAN_TEST_OUTPUT_EXTRA_SPAM:
```

Comment:
You'll want to add `# nocoverage` on this function, I think, since one path or the other is going to be not run in every individual test run.